### PR TITLE
Added support for OAuth2 permissions in resource application.

### PR DESCRIPTION
**Pull request by jorgecarleitao**
See #162 for details.

### CODE REVIEW

**Comment by jorgecarleitao** [comment]
I've tested both create, destroy, and change of the resource. The config I used was

```
provider "azuread" {

  subscription_id = "..."
  client_id       = "..."
  client_secret   = "..."
  tenant_id       = "..."
}

# Create an application
resource "azuread_application" "example" {
  name = "ExampleApp"

  oauth2_permissions {
      admin_consent_description  = "example"
      admin_consent_display_name = "example"
      type                       = "Admin"
      value                      = "access"
  }

  oauth2_permissions {
      admin_consent_description  = "example"
      admin_consent_display_name = "example"
      type                       = "User"
      value                      = "write"
  }
}
```

and 
1. `terraform apply -auto-approve`
2. erased one of the items and `terraform apply -auto-approve`
3. added the item back and `terraform apply -auto-approve`
4. changed value of second item from `write` to `read` and `terraform apply -auto-approve`
5. `terraform destroy -auto-approve`

I validated each change both on the local state and on the portal.

File: azuread/resource_application.go [hunk 1]
```diff
@@ -202,7 +202,61 @@ func resourceApplication() *schema.Resource {
 				Computed: true,
 			},
 
-			"oauth2_permissions": graph.SchemaOauth2PermissionsComputed(),
+			"oauth2_permissions": {
+				Type:     schema.TypeSet,
+				Optional: true,
```
Comment (ennui93) [inline azuread/resource_application.go]: Per [katbyte's comment](https://github.com/terraform-providers/terraform-provider-azuread/issues/162#issuecomment-543807695) and the other discussion in #162, should this also be computed?

File: website/docs/r/application.html.markdown [hunk 2]
```diff
@@ -125,6 +127,24 @@ The following arguments are supported:
 
 * `value` - (Optional) Specifies the value of the roles claim that the application should expect in the authentication and access tokens.
 
+---
+
+`oauth2_permission` supports the following:
```
Comment (ennui93) [inline website/docs/r/application.html.markdown]:
```suggestion
`oauth2_permissions` supports the following:
```

**Comment by stevebering** [comment]
I would really like to see this merged and released as it is preventing me from using Terraform to automate our App Registrations.

**Comment by PeterStephenson** [comment]
We're currently also getting stuck by this, is there an ETA for the feature or anything we can do to help finish it?

**Comment by MetalGuardian** [comment]
I'd also like to see this merge. Ho can we make it happen?

**Comment by katbyte** [comment]
@MetalGuardian - iirc there needs to be a good ux for leaving it empty (probably marking it as computed) and then changing/adding/updating new permissions. i pulled it locally and tried a few things quickly a couple months ago but nothing stuck. So the PR basically needs to be update as per above comments & then tweaked to have a better UX around the default values.

**Comment by manicminer** [comment]
Closing in favor of #252

**Pull request closed without merging**